src/nodes/ImageView.vue

```diff
@@ -280,18 +248,59 @@ export default {
 			this.loaded = true
 			return
 		}
-		const img = new Image()
-		img.onload = () => {
-			this.imageLoaded = true
-		}
-		img.onerror = () => {
-			this.failed = true
-			this.imageLoaded = false
-			this.loaded = true
-		}
-		img.src = this.imageUrl
+		this.init()
 	},
 	methods: {
+		init() {
+			if (this.src.startsWith('text://')) {
+				const imageFileName = getQueryVariable(this.src, 'imageFileName')
+				this.loadImage(this.getTextApiUrl(imageFileName))
+				return
+			}
+			if (this.src.startsWith(`.attachments.${this.currentSession?.documentId}/`)) {
+				const imageFileName = decodeURIComponent(this.src.replace(`.attachments.${this.currentSession?.documentId}/`, '').split('?')[0])
+				this.loadImage(this.getTextApiUrl(imageFileName))
+				return
+			}
+			if (this.isRemoteUrl || this.isPreviewUrl || this.isDataUrl) {
```

Comment:
Can we turn it into a computed?
```suggestion
			if (this.isDirectUrl) {
```

Reply:
Yep!